drivers/timer/ite_it8xxx2_timer.c

```diff
@@ -0,0 +1,327 @@
+	ECReg(EXT_CTL_B + ((idx) * 8)) |= BIT(0);
+	ECReg(EXT_LH3R_B + ((idx) * 8)) = (((cnt) >> 24) & 0xFF);
+	ECReg(EXT_LH2R_B + ((idx) * 8)) = (((cnt) >> 16) & 0xFF);
+	ECReg(EXT_LHR_B  + ((idx) * 8)) = (((cnt) >>  8) & 0xFF);
+	ECReg(EXT_LLR_B  + ((idx) * 8)) = (((cnt) >>  0) & 0xFF);
+}
+
+/* The following function:
+ * disable, enable, check_flag, clear_flag, wait,
+ * can be used only for Timer #3 ~ #7
+ */
+static void ite_timer_disable(uint8_t idx)
+{
+	CLEAR_MASK_EXT(IER19, (BIT(3 + (idx))));
+}
+
+static void ite_timer_enable(uint8_t idx)
+{
+	SET_MASK_EXT(IER19, (BIT(3 + (idx))));
+}
+
+static void ite_timer_clear_flag(uint8_t idx)
+{
+	ISR19 = BIT(3 + (idx));
+}
+
+static uint8_t ite_timer_wait(uint8_t idx)
+{
+	return (!(ISR19 & BIT(3 + (idx))));
+}
+
+/**
+ * timer_init()
+ */
+int timer_init(uint8_t idx, uint8_t psr, uint8_t initial_state,
+	       uint8_t enable_isr, uint32_t cnt)
+{
+	/* Setup Triggered Mode -> Rising-Edge Trig. */
+	if (idx != EXT_TIMER_8) {
+		IELMR19 |= BIT(3 + idx);
+		IPOLR19 &= (~(BIT(3 + idx)));
+	} else {
+		IELMR10 |= BIT(0);
+		IPOLR10 &= (~(BIT(0)));
+	}
+
+	/* Setup prescaler */
+	ECReg(EXT_PSC_B + (8 * idx)) = psr;
+
+	/* Reload counter */
+	ite_timer_reload(idx, cnt);
+
+	/* Start counting or not */
+	if (initial_state) {
+		ite_timer_restart(idx);
+	} else {
+		ite_timer_stop(idx);
+	}
+
+	/* Enable ISR or not & Clear flag */
+	if (idx != EXT_TIMER_8) {
+		if (enable_isr) {
+			ite_timer_enable(idx);
+		} else {
+			ite_timer_disable(idx);
+		}
+		ite_timer_clear_flag(idx);
+	} else {
+		if (enable_isr) {
+			SET_MASK_EXT(IER10, BIT(0));
+		} else {
+			CLEAR_MASK_EXT(IER10, BIT(0));
+		}
+		ISR10 = BIT(0);
+	}
+	return 0;
+}
+
+int timer_init_ms(uint8_t idx, uint8_t psr, uint8_t initial_state,
+		  uint8_t enable_isr, uint32_t u32MilliSec)
+{
+	uint32_t cnt;
+
+	if (psr == ET_PSR_32K) {
+		cnt = MILLI_SEC_TO_COUNT(32768, u32MilliSec);
+	} else if (psr == ET_PSR_1K) {
+		cnt = MILLI_SEC_TO_COUNT(1024, u32MilliSec);
+	} else if (psr == ET_PSR_32) {
+		cnt = MILLI_SEC_TO_COUNT(32, u32MilliSec);
+	} else if (psr == ET_PSR_8M) {
+		cnt = u32MilliSec * 8000; /* fixed overflow issue */
+	} else {
+		return -1;
+	}
+
+	/* 24-bits only */
+	if (cnt >> 24) {
+		return -2;
+	}
+	return timer_init(idx, psr, initial_state, enable_isr, cnt);
+}
+
+int timer_init_us(uint8_t idx, uint8_t psr,
+		  uint8_t initial_state, uint8_t enable_isr,
+		  uint32_t u32MicroSec)
+{
+	uint32_t cnt;
+
+	if (psr == ET_PSR_32K) {
+		cnt = MICRO_SEC_TO_COUNT(32768, u32MicroSec);
+	} else if (psr == ET_PSR_1K) {
+		cnt = MICRO_SEC_TO_COUNT(1024, u32MicroSec);
+	} else if (psr == ET_PSR_32) {
+		cnt = MICRO_SEC_TO_COUNT(32, u32MicroSec);
+	} else if (psr == ET_PSR_8M) {
+		cnt = u32MicroSec * 8; /* fixed overflow issue */
+	} else {
+		return -1;
+	}
+
+	/* 24-bits only */
+	if (cnt >> 24) {
+		return -2;
+	}
+	return timer_init(idx, psr, initial_state, enable_isr, cnt);
+}
+
+void timer_init_combine(uint8_t idx, uint8_t bEnable)
```

Comment:
could probably make several of the functions in this file static

Reply:
OK. And I remove some function which were not being used.